src/lib/ffi/ffi.h

```diff
@@ -2196,6 +2204,182 @@ int botan_x509_cert_verify(int* validation_result,
+typedef struct botan_x509_time_struct* botan_x509_time_t;
+typedef struct botan_x509_ext_as_blocks_struct* botan_x509_ext_as_blocks_t;
+typedef struct botan_x509_ext_ip_addr_blocks_struct* botan_x509_ext_ip_addr_blocks_t;
+typedef struct botan_x509_ca_struct* botan_x509_ca_t;
+typedef struct botan_x509_pkcs10_req_struct* botan_x509_pkcs10_req_t;
+
+BOTAN_FFI_EXPORT(3, 9) int botan_x509_cert_opts_destroy(botan_x509_cert_opts_t opts);
+
+BOTAN_FFI_EXPORT(3, 9) int botan_x509_time_destroy(botan_x509_time_t time);
+
+BOTAN_FFI_EXPORT(3, 9) int botan_x509_ext_ip_addr_blocks_destroy(botan_x509_ext_ip_addr_blocks_t ip_addr_blocks);
+
+BOTAN_FFI_EXPORT(3, 9) int botan_x509_ext_as_blocks_destroy(botan_x509_ext_as_blocks_t as_blocks);
+
+BOTAN_FFI_EXPORT(3, 9) int botan_x509_ca_destroy(botan_x509_ca_t ca);
+
+BOTAN_FFI_EXPORT(3, 9) int botan_x509_pkcs10_req_destroy(botan_x509_pkcs10_req_t req);
+
+BOTAN_FFI_EXPORT(3, 9)
+int botan_x509_create_cert_opts(botan_x509_cert_opts_t* opts_obj, const char* opts, uint32_t* expire_time);
+
+BOTAN_FFI_EXPORT(3, 9) int botan_x509_cert_opts_common_name(botan_x509_cert_opts_t opts, const char* name);
+
+BOTAN_FFI_EXPORT(3, 9) int botan_x509_cert_opts_country(botan_x509_cert_opts_t opts, const char* country);
+
+BOTAN_FFI_EXPORT(3, 9) int botan_x509_cert_opts_organization(botan_x509_cert_opts_t opts, const char* organization);
+
+BOTAN_FFI_EXPORT(3, 9) int botan_x509_cert_opts_org_unit(botan_x509_cert_opts_t opts, const char* org_unit);
+
+BOTAN_FFI_EXPORT(3, 9) int botan_x509_cert_opts_locality(botan_x509_cert_opts_t opts, const char* locality);
+
+BOTAN_FFI_EXPORT(3, 9) int botan_x509_cert_opts_state(botan_x509_cert_opts_t opts, const char* state);
+
+BOTAN_FFI_EXPORT(3, 9) int botan_x509_cert_opts_serial_number(botan_x509_cert_opts_t opts, const char* serial_number);
+
+BOTAN_FFI_EXPORT(3, 9) int botan_x509_cert_opts_email(botan_x509_cert_opts_t opts, const char* email);
+
+BOTAN_FFI_EXPORT(3, 9) int botan_x509_cert_opts_uri(botan_x509_cert_opts_t opts, const char* uri);
+
+BOTAN_FFI_EXPORT(3, 9) int botan_x509_cert_opts_ip(botan_x509_cert_opts_t opts, const char* ip);
+
+BOTAN_FFI_EXPORT(3, 9) int botan_x509_cert_opts_dns(botan_x509_cert_opts_t opts, const char* dns);
+
+BOTAN_FFI_EXPORT(3, 9) int botan_x509_cert_opts_xmpp(botan_x509_cert_opts_t opts, const char* xmpp);
+
+BOTAN_FFI_EXPORT(3, 9) int botan_x509_cert_opts_challenge(botan_x509_cert_opts_t opts, const char* challenge);
+
+BOTAN_FFI_EXPORT(3, 9)
+int botan_x509_cert_opts_more_org_units(botan_x509_cert_opts_t opts, const char** more_org_units, size_t cnt);
+
+BOTAN_FFI_EXPORT(3, 9)
+int botan_x509_cert_opts_more_dns(botan_x509_cert_opts_t opts, const char** more_dns, size_t cnt);
+
+BOTAN_FFI_EXPORT(3, 9) int botan_x509_cert_opts_ca_key(botan_x509_cert_opts_t opts, size_t limit);
+
+BOTAN_FFI_EXPORT(3, 9) int botan_x509_cert_opts_padding_scheme(botan_x509_cert_opts_t opts, const char* scheme);
```

Comment:
Let's remove this one entirely. The padding scheme shouldn't have been part of that struct in the first place. (It also is *only* relevant for RSA and the desire to use RSA-PSS certs).